### PR TITLE
Extensible Gatsby directives

### PR DESCRIPTION
**Pull request by pmelab**
## Package(s) involved

`@amazeelabs/gatsby-source-silverback`

## Description of changes

Allow new directives to be added to Gatsby's GraphQL schema.

## Motivation and context

Provide reusable directives for common use cases.

## How has this been tested?

- [x] Manually
- [x] Unit tests
- [x] Integration tests

### CODE REVIEW

**Comment by pmelab** [comment]
Current dependencies on/for this PR:
* development
  * **PR #1418** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1418" target="_blank">Graphite</a> 
    * **PR #1420** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1420" target="_blank">Graphite</a>  👈
      * **PR #1422** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1422" target="_blank">Graphite</a> 
        * **PR #1423** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1423" target="_blank">Graphite</a> 
          * **PR #1424** <a href="https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1424" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1420?utm_source=stack-comment).

File: packages/npm/@amazeelabs/gatsby-source-silverback/src/helpers/schema.ts [hunk 1]
```diff
@@ -108,3 +144,46 @@ export function cleanSchema(schema: DocumentNode) {
...
+export function processDirectiveArguments(
+  parent: unknown,
+  args: Record<string, any>,
+  spec: Record<string, unknown>,
+) {
+  return Object.fromEntries(
+    Object.keys(spec).map((key) => {
+      const val = spec[key];
+      if (isString(val)) {
+        if (val === '$') {
```
Comment (Leksat) [inline packages/npm/@amazeelabs/gatsby-source-silverback/src/helpers/schema.ts:157]: Can't see `$` magic explained in the readme. At least not in this PR.

I know it's documented in `amazeelabs/graphql_directives`. But that one is from another world.

File: packages/npm/@amazeelabs/gatsby-source-silverback/test/schema/schema.graphql [hunk 2]
```diff
@@ -1,6 +1,9 @@
 scalar Email
 
 type Query {
+  value: String @echo(msg: "value from schema")
+  argument(msg: String!): String @echo(msg: "$msg")
+  parent: String @echo(msg: "parent value") @echo(msg: "$")
```
Comment (Leksat) [inline packages/npm/@amazeelabs/gatsby-source-silverback/test/schema/schema.graphql]: Holy moly 🤑 Black magic 🙈

**Review by Leksat — APPROVED** [review]

File: packages/npm/@amazeelabs/gatsby-source-silverback/src/helpers/schema.ts (hunk 1 above)
Comment (pmelab) [reply packages/npm/@amazeelabs/gatsby-source-silverback/src/helpers/schema.ts:157]: I added a reference to the readme.

File: packages/npm/@amazeelabs/gatsby-source-silverback/test/schema/schema.graphql (hunk 2 above)
Comment (pmelab) [reply packages/npm/@amazeelabs/gatsby-source-silverback/test/schema/schema.graphql]: Any sufficiently advanced technology is indistinguishable from magic.

**Comment by pmelab** [comment]
### Merge Activity

* **Sep 27, 6:18 AM**: @pmelab started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1420).
* **Sep 27, 6:19 AM**: [Graphite](https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1420) rebased this pull request as part of a merge.
* **Sep 27, 6:30 AM**: [Graphite](https://app.graphite.dev/github/pr/AmazeeLabs/silverback-mono/1420) couldn't merge this PR because it had conflicts with the trunk branch.

**Pull request merged**